runtime/src/accounts.rs

```diff
@@ -356,6 +367,11 @@ impl Accounts {
                                             load_zero_lamports,
                                         )
                                     {
+                                        Self::accumulate_and_check_transaction_accounts_data_size(
+                                            &mut accumulated_accounts_data_size,
+                                            programdata_account.data().len(),
+                                            max_accounts_data_size_by_requested_cu,
+                                        )?;
```

Comment:
Unfortunately we can't make consensus-impacting decisions based on the size of the program data account right now because of the way `ExtendProgramData` was implemented. `ExtendProgramData` doesn't require a write lock on the program account so it could update the length of the account in parallel with a transaction that is very close to the data size limit.

I suggest that we modify `ExtendProgramData` to require a write lock on the program account in a separate PR since I think the size of program data should impact the data size limit.

Reply:
> I suggest that we modify `ExtendProgramData` to require a write lock on the program account in a separate PR since I think the size of program data should impact the data size limit.

I have a PR to resolve this here: https://github.com/solana-labs/solana/pull/27911